### PR TITLE
Implement singleton semantics for CCEServer

### PR DESCRIPTION
**Pull request by PaulBrachmann**

### CODE REVIEW

File: packages/CollabCodeEdit-Server.package/CCEServer.class/class/breakdown.st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+breakdown
+breakdown
+	self currentInstance ifNotNil: [
+		self currentInstance breakdown.
+		self currentInstance: nil
+	]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/class/breakdown.st]: Blocks should be opened and closed in the same line.

File: packages/CollabCodeEdit-Server.package/CCEServer.class/class/start.st [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
 
-	^ self new start
\ No newline at end of file
+	^ self currentInstance ifNil: [
+		| newInstance |
+
+		newInstance := self new start.
+		self currentInstance: newInstance.
+		^ newInstance
+	]
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/class/start.st]: Blocks should be opened and closed in the same line.

File: packages/CollabCodeEdit-Server.package/CCEServer.class/properties.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
 	"classinstvars" : [
 		 ],
 	"classvars" : [
-		 ],
+		"CurrentInstance" ],
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/properties.json:6]: Variables should not be capitalized.

File: packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testBreakdown.st [hunk 4]
```diff
@@ -0,0 +1,10 @@
+tests
+testBreakdown
+	CCEServer
+		breakdown;
+		start.
+	CCEServer currentInstance ifNil: [ self fail ].
+	
+	CCEServer breakdown.
+	CCEServer currentInstance ifNotNil: [ self fail ]
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testBreakdown.st]: Trailing whitespace is not permited by the styleguide.

File: packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	3 timesRepeat: [
+		self assert: server equals: CCEServer currentInstance.
+
+		server := CCEServer restart.
+		server ifNil: [ self fail ].
+		self assert: CCEServer equals: server class.
+		self deny: previousServer equals: server.
+
+		previousServer := server ]
+
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st]: Trailing whitespace is not permited by the styleguide.

File: packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testStart.st [hunk 6]
```diff
@@ -0,0 +1,11 @@
+tests
+testStart
+	| server |
+	server := nil.
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testStart.st:5]: After declaration of tmp variables a line should follow.

File: packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st [hunk 7]
```diff
@@ -0,0 +1,16 @@
+tests
+testRestart
+	| server previousServer |
+	server := nil.
+	previousServer := nil.
+
+	3 timesRepeat: [
```
Comment (arminWel) [inline packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st:8]: Blocks should be opened and closed in the same line.

**Review by arminWel — CHANGES_REQUESTED** [review]

File: packages/CollabCodeEdit-Server.package/CCEServer.class/properties.json (hunk 3 above)
Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/properties.json:6]: Class variables should be.

File: packages/CollabCodeEdit-Server.package/CCEServer.class/class/start.st [hunk 8]
```diff
@@ -1,5 +1,10 @@
 startup
 start
```
Comment (arminWel) [inline packages/CollabCodeEdit-Server.package/CCEServer.class/class/start.st:2]: Maybe a more wordy method name instead of a comment.

Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Server.package/CCEServer.class/class/start.st:2]: Good suggestion. For the time being, I would, however, like to keep it like as is.
The naming was a conscious decision made in an effort to stay consistent with the instances' interface.

File: packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st (hunk 7 above)
Comment (PaulBrachmann) [reply packages/CollabCodeEdit-Tests.package/CCEServerClassTest.class/instance/testRestart.st:8]: After a brief discussion, the style guide has been updated in favor of the above formatting.

**Pull request merged**